src/Functions/FunctionsFormatting.h

```diff
@@ -278,4 +279,210 @@ class FunctionFormatReadableQuantity : public IFunction
+
+    bool isVariadic() const override { return true; }
+
+    size_t getNumberOfArguments() const override { return 0; }
+
+    DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
+    {
+        if (arguments.size() < 1)
+            throw Exception(
+                "Number of arguments for function " + getName() + " doesn't match: passed " + toString(arguments.size())
+                    + ", should be at least 1.",
+                ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);
+
+        if (arguments.size() > 2)
+            throw Exception(
+                "Number of arguments for function " + getName() + " doesn't match: passed " + toString(arguments.size())
+                    + ", should be at most 2.",
+                ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);
+
+        const IDataType & type = *arguments[0];
+
+        if (!isNativeNumber(type))
+            throw Exception("Cannot format " + type.getName() + " as time delta", ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+
+        if (arguments.size() == 2)
+        {
+            const auto * maximum_unit_arg = arguments[1].get();
+            if (!isStringOrFixedString(maximum_unit_arg))
+                throw Exception("Illegal type " + maximum_unit_arg->getName() + " of argument maximum_unit of function "
+                                + getName(), ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+        }
+
+        return std::make_shared<DataTypeString>();
+    }
+
+    ColumnNumbers getArgumentsThatAreAlwaysConstant() const override { return {1}; }
+
+    bool useDefaultImplementationForConstants() const override { return true; }
+
+    void executeImpl(Block & block, const ColumnNumbers & arguments, size_t result, size_t /*input_rows_count*/) const override
+    {
+        if (!(executeType<UInt8>(block, arguments, result)
+            || executeType<UInt16>(block, arguments, result)
+            || executeType<UInt32>(block, arguments, result)
+            || executeType<UInt64>(block, arguments, result)
+            || executeType<Int8>(block, arguments, result)
+            || executeType<Int16>(block, arguments, result)
+            || executeType<Int32>(block, arguments, result)
+            || executeType<Int64>(block, arguments, result)
+            || executeType<Float32>(block, arguments, result)
+            || executeType<Float64>(block, arguments, result)))
+            throw Exception("Illegal column " + block.getByPosition(arguments[0]).column->getName()
+                + " of argument of function " + getName(),
+                ErrorCodes::ILLEGAL_COLUMN);
+    }
+
+private:
+
+    void formatReadableTimeDelta(double value, String maximum_unit, DB::WriteBuffer & out) const
+    {
+        // 60 SECONDS       = 1 MINUTE
+        // 3600 SECONDS     = 1 HOUR
+        // 86400 SECONDS    = 1 DAY
+        // 2635200 SECONDS  = 30.5 DAYS = 1 MONTH
+        // 31536000 SECONDS = 365 DAYS  = 1 YEAR
+
+        char sig = value<0?-1:1;
```

Comment:
When compiling for ARM, the error arise:

```
/build/src/Functions/FunctionsFormatting.h:443:17: error: result of comparison of unsigned expression < 0 is always false [-Werror,-Wtautological-unsigned-zero-compare]
        if (sig < 0)
            ~~~ ^ ~
```

Because `char` is unsigned on ARM.
Better to use `int8_t`.